pkg/ocm/connection.go

```diff
@@ -109,11 +111,31 @@ func (b *ConnectionBuilder) Build() (result *sdk.Connection, err error) {
 	}
 	builder.Insecure(b.cfg.Insecure)
 
-	// Create the connection:
-	result, err = builder.Build()
+	result, err = b.buildConnection(builder)
 	if err != nil {
 		return
 	}
+	access, _, err := result.Tokens()
```

Comment:
The `Tokens` method receives an optional `expiresIn` parameter that tells it to make sure that the token needs to have at least that life remaining. For example, to make sure that you get a token that will live for at least 10 minutes you can do this:

```go
accessToken, _, err := result.Tokens(10 * time.Minutes)
```

If the current access token already had that life time remaining then it will not request a new one, and you will be sure that you can use it for the next 10 minutes. If it doesn't have that live time remaining then a new one will be requested, and if there is an ongoing SSO outage it will fail. So there only two possible outcomes:

1. You get a token that is valid for at least the next 10 minutes.
2. You get an error, which you can assume was an SSO outage.

So I think you don't need to create the connection again. You only need to check if `err != nil` and then report something like `Can't create connection because wasn't able to get an authentication token`.

Reply:
@jhernand  please take a look and let me know how it looks now

Reply:
Looks good to me.

(You will have to run `go fmt` on that file.)